The DriverClassName is com.mysql.cj.jdbc.Driver ShardingSphere-Proxy start failed.

### Which version of ShardingSphere did you use?

`master`

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

`Sharding-Proxy`

```
[ERROR] 2022-01-07 19:12:53.820 [main] com.zaxxer.hikari.HikariConfig - Failed to load driver class com.mysql.cj.jdbc.Driver from HikariConfig class classloader sun.misc.Launcher$AppClassLoader@18b4aac2
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.shardingsphere.infra.config.datasource.pool.creator.DataSourceReflection.setField(DataSourceReflection.java:133)
	at org.apache.shardingsphere.infra.config.datasource.pool.creator.DataSourceReflection.lambda$setField$0(DataSourceReflection.java:118)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.apache.shardingsphere.infra.config.datasource.pool.creator.DataSourceReflection.setField(DataSourceReflection.java:118)
	at org.apache.shardingsphere.infra.config.datasource.pool.creator.DataSourcePoolCreator.setConfiguredFields(DataSourcePoolCreator.java:110)
	at org.apache.shardingsphere.infra.config.datasource.pool.creator.DataSourcePoolCreator.createDataSource(DataSourcePoolCreator.java:82)
	at org.apache.shardingsphere.infra.config.datasource.pool.creator.DataSourcePoolCreatorUtil.getDataSource(DataSourcePoolCreatorUtil.java:43)
	at org.apache.shardingsphere.mode.manager.cluster.ClusterContextManagerBuilder.getLoadDataSources(ClusterContextManagerBuilder.java:187)
	at org.apache.shardingsphere.mode.manager.cluster.ClusterContextManagerBuilder.loadDataSourcesMap(ClusterContextManagerBuilder.java:162)
	at org.apache.shardingsphere.mode.manager.cluster.ClusterContextManagerBuilder.beforeBuildContextManager(ClusterContextManagerBuilder.java:100)
	at org.apache.shardingsphere.mode.manager.cluster.ClusterContextManagerBuilder.build(ClusterContextManagerBuilder.java:84)
	at org.apache.shardingsphere.proxy.initializer.BootstrapInitializer.createContextManager(BootstrapInitializer.java:84)
	at org.apache.shardingsphere.proxy.initializer.BootstrapInitializer.init(BootstrapInitializer.java:71)
	at org.apache.shardingsphere.proxy.Bootstrap.main(Bootstrap.java:47)
Caused by: java.lang.RuntimeException: Failed to load driver class com.mysql.cj.jdbc.Driver in either of HikariConfig class loader or Thread context classloader
	at com.zaxxer.hikari.HikariConfig.setDriverClassName(HikariConfig.java:485)
	... 18 more
```
### Zookeeper Configuration

```
ds0:
  initializationFailTimeout: 1
  validationTimeout: 5000
  readOnly: false
  registerMbeans: false
  isolateInternalQueries: false
  maxLifetime: 1800000
  leakDetectionThreshold: 0
  running: false
  password: '123456'
  allowPoolSuspension: false
  idleTimeout: 600000
  driverClassName: com.mysql.cj.jdbc.Driver
  jdbcUrl: jdbc:mysql://127.0.0.1:3306/test0
  closed: false
  dataSourceClassName: com.zaxxer.hikari.HikariDataSource
  autoCommit: true
  connectionTimeout: 30000
  username: root
ds1:
  initializationFailTimeout: 1
  validationTimeout: 5000
  readOnly: false
  registerMbeans: false
  isolateInternalQueries: false
  maxLifetime: 1800000
  leakDetectionThreshold: 0
  running: false
  password: '123456'
  allowPoolSuspension: false
  idleTimeout: 600000
  driverClassName: com.mysql.cj.jdbc.Driver
  jdbcUrl: jdbc:mysql://127.0.0.1:3306/test1
  closed: false
  dataSourceClassName: com.zaxxer.hikari.HikariDataSource
  autoCommit: true
  connectionTimeout: 30000
  username: root
ds2:
  initializationFailTimeout: 1
  validationTimeout: 5000
  readOnly: false
  registerMbeans: false
  isolateInternalQueries: false
  maxLifetime: 1800000
  leakDetectionThreshold: 0
  running: false
  password: '123456'
  allowPoolSuspension: false
  idleTimeout: 600000
  driverClassName: com.mysql.cj.jdbc.Driver
  jdbcUrl: jdbc:mysql://127.0.0.1:3306/test1
  closed: false
  dataSourceClassName: com.zaxxer.hikari.HikariDataSource
  autoCommit: true
  connectionTimeout: 30000
  username: root

```

--- a/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/pool/creator/DataSourceReflection.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/pool/creator/DataSourceReflection.java
@@ -43,7 +43,7 @@ public final class DataSourceReflection {
     
     static {
         GENERAL_CLASS_TYPES = Sets.newHashSet(boolean.class, Boolean.class, int.class, Integer.class, long.class, Long.class, String.class, Collection.class, List.class, Properties.class);
-        SKIPPED_PROPERTY_KEYS = Sets.newHashSet("loginTimeout");
+        SKIPPED_PROPERTY_KEYS = Sets.newHashSet("loginTimeout", "driverClassName");
     }
     
     private static final Collection<Class<?>> GENERAL_CLASS_TYPES;

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/pool/creator/DataSourcePoolCreatorTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/pool/creator/DataSourcePoolCreatorTest.java
@@ -47,13 +47,11 @@ public final class DataSourcePoolCreatorTest {
     @Test
     public void assertCreate() {
         MockedDataSource actual = (MockedDataSource) DataSourcePoolCreator.create(new DataSourceProperties(MockedDataSource.class.getName(), createProperties()));
-        assertThat(actual.getDriverClassName(), is(MockedDataSource.class.getName()));
         assertDataSource(actual);
     }
     
     private Map<String, Object> createProperties() {
         Map<String, Object> result = new LinkedHashMap<>();
-        result.put("driverClassName", MockedDataSource.class.getName());
         result.put("url", "jdbc:mock://127.0.0.1/foo_ds");
         result.put("username", "root");
         result.put("password", "root");
@@ -75,7 +73,6 @@ public final class DataSourcePoolCreatorTest {
     
     private Map<String, Object> createDefaultProperties() {
         Map<String, Object> result = new HashMap<>();
-        result.put("driverClassName", MockedDataSource.class.getName());
         result.put("url", "jdbc:mock://127.0.0.1/foo_ds");
         result.put("username", "root");
         result.put("password", "root");

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/pool/creator/type/DBCPDataSourcePoolCreatorTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/pool/creator/type/DBCPDataSourcePoolCreatorTest.java
@@ -38,7 +38,6 @@ public final class DBCPDataSourcePoolCreatorTest {
     public void assertCreateDataSource() {
         BasicDataSource actual = (BasicDataSource) DataSourcePoolCreator.create(new DataSourceProperties(BasicDataSource.class.getName(), createDataSourceProperties()));
         assertThat(actual.getUrl(), is("jdbc:mock://127.0.0.1/foo_ds"));
-        assertThat(actual.getDriverClassName(), is(MockedDataSource.class.getName()));
         assertThat(actual.getUsername(), is("root"));
         assertThat(actual.getPassword(), is("root"));
         assertThat(getConnectionProperties(actual), is(createJdbcUrlProperties()));

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/pool/creator/type/HikariDataSourcePoolCreatorTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/pool/creator/type/HikariDataSourcePoolCreatorTest.java
@@ -36,7 +36,6 @@ public final class HikariDataSourcePoolCreatorTest {
     public void assertCreateDataSource() {
         HikariDataSource actual = (HikariDataSource) DataSourcePoolCreator.create(new DataSourceProperties(HikariDataSource.class.getName(), createDataSourceProperties()));
         assertThat(actual.getJdbcUrl(), is("jdbc:mock://127.0.0.1/foo_ds"));
-        assertThat(actual.getDriverClassName(), is(MockedDataSource.class.getName()));
         assertThat(actual.getUsername(), is("root"));
         assertThat(actual.getPassword(), is("root"));
         assertThat(actual.getDataSourceProperties(), is(createJdbcUrlProperties()));

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/pool/creator/type/TomcatDBCPDataSourcePoolCreatorTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/pool/creator/type/TomcatDBCPDataSourcePoolCreatorTest.java
@@ -20,7 +20,6 @@ package org.apache.shardingsphere.infra.datasource.pool.creator.type;
 import lombok.SneakyThrows;
 import org.apache.shardingsphere.infra.datasource.pool.creator.DataSourcePoolCreator;
 import org.apache.shardingsphere.infra.datasource.props.DataSourceProperties;
-import org.apache.shardingsphere.test.mock.MockedDataSource;
 import org.apache.tomcat.dbcp.dbcp2.BasicDataSource;
 import org.junit.Test;
 
@@ -38,7 +37,6 @@ public final class TomcatDBCPDataSourcePoolCreatorTest {
     public void assertCreateDataSource() {
         BasicDataSource actual = (BasicDataSource) DataSourcePoolCreator.create(new DataSourceProperties(BasicDataSource.class.getName(), createDataSourceProperties()));
         assertThat(actual.getUrl(), is("jdbc:mock://127.0.0.1/foo_ds"));
-        assertThat(actual.getDriverClassName(), is(MockedDataSource.class.getName()));
         assertThat(actual.getUsername(), is("root"));
         assertThat(actual.getPassword(), is("root"));
         assertThat(getConnectionProperties(actual), is(createJdbcUrlProperties()));
@@ -47,7 +45,6 @@ public final class TomcatDBCPDataSourcePoolCreatorTest {
     private Map<String, Object> createDataSourceProperties() {
         Map<String, Object> result = new HashMap<>();
         result.put("url", "jdbc:mock://127.0.0.1/foo_ds");
-        result.put("driverClassName", MockedDataSource.class.getName());
         result.put("username", "root");
         result.put("password", "root");
         result.put("connectionProperties", createJdbcUrlProperties());

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesTest.java
@@ -55,7 +55,6 @@ public final class DataSourcePropertiesTest {
         actualDataSource.setConnectionInitSqls(Arrays.asList("set names utf8mb4;", "set names utf8;"));
         DataSourceProperties actual = DataSourcePropertiesCreator.create(actualDataSource);
         assertThat(actual.getDataSourceClassName(), is(BasicDataSource.class.getName()));
-        assertThat(actual.getAllLocalProperties().get("driverClassName").toString(), is(MockedDataSource.class.getName()));
         assertThat(actual.getAllLocalProperties().get("url").toString(), is("jdbc:mock://127.0.0.1/foo_ds"));
         assertThat(actual.getAllLocalProperties().get("username").toString(), is("root"));
         assertThat(actual.getAllLocalProperties().get("password").toString(), is("root"));

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidatorTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesValidatorTest.java
@@ -19,7 +19,6 @@ package org.apache.shardingsphere.infra.datasource.props;
 
 import com.zaxxer.hikari.HikariDataSource;
 import org.apache.shardingsphere.infra.distsql.exception.resource.InvalidResourcesException;
-import org.apache.shardingsphere.test.mock.MockedDataSource;
 import org.junit.Test;
 
 import java.util.Collections;
@@ -35,7 +34,6 @@ public final class DataSourcePropertiesValidatorTest {
     
     private Map<String, Object> createValidProperties() {
         Map<String, Object> result = new HashMap<>();
-        result.put("driverClassName", MockedDataSource.class.getName());
         result.put("jdbcUrl", "jdbc:mock://127.0.0.1/foo_ds");
         result.put("username", "root");
         result.put("password", "root");
@@ -49,7 +47,7 @@ public final class DataSourcePropertiesValidatorTest {
     
     private Map<String, Object> createInvalidProperties() {
         Map<String, Object> result = new HashMap<>();
-        result.put("driverClassName", "InvalidDriver");
+        result.put("jdbcUrl", "InvalidJdbcUrl");
         return result;
     }
 }

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/yaml/config/swapper/YamlDataSourcePropertiesSwapperTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/yaml/config/swapper/YamlDataSourcePropertiesSwapperTest.java
@@ -39,12 +39,10 @@ public final class YamlDataSourcePropertiesSwapperTest {
         Map<String, Map<String, Object>> yamlConfig = createYamlConfig();
         Map<String, DataSource> dataSources = swapper.swapToDataSources(yamlConfig);
         HikariDataSource actual0 = (HikariDataSource) dataSources.get("ds_0");
-        assertThat(actual0.getDriverClassName(), is(MockedDataSource.class.getName()));
         assertThat(actual0.getJdbcUrl(), is("jdbc:mock://127.0.0.1/ds_0"));
         assertThat(actual0.getUsername(), is("root"));
         assertThat(actual0.getPassword(), is("root"));
         HikariDataSource actual1 = (HikariDataSource) dataSources.get("ds_1");
-        assertThat(actual1.getDriverClassName(), is(MockedDataSource.class.getName()));
         assertThat(actual1.getJdbcUrl(), is("jdbc:mock://127.0.0.1/ds_1"));
         assertThat(actual1.getUsername(), is("root"));
         assertThat(actual1.getPassword(), is("root"));
@@ -57,7 +55,6 @@ public final class YamlDataSourcePropertiesSwapperTest {
         yamlConfig.put("url", "xx:xxx");
         yamlConfig.put("username", "root");
         DataSourceProperties actual = swapper.swapToDataSourceProperties(yamlConfig);
-        assertThat(actual.getDataSourceClassName(), is(MockedDataSource.class.getName()));
         assertThat(actual.getAllLocalProperties().size(), is(2));
         assertThat(actual.getAllLocalProperties().get("url").toString(), is("xx:xxx"));
         assertThat(actual.getAllLocalProperties().get("username").toString(), is("root"));
@@ -88,7 +85,6 @@ public final class YamlDataSourcePropertiesSwapperTest {
     private Map<String, Object> createPropertyMap(final String name) {
         Map<String, Object> result = new LinkedHashMap<>(5, 1);
         result.put("dataSourceClassName", "com.zaxxer.hikari.HikariDataSource");
-        result.put("driverClassName", MockedDataSource.class.getName());
         result.put("jdbcUrl", String.format("jdbc:mock://127.0.0.1/%s", name));
         result.put("username", "root");
         result.put("password", "root");

--- a/shardingsphere-mode/shardingsphere-mode-core/src/test/java/org/apache/shardingsphere/mode/metadata/persist/service/impl/DataSourcePersistServiceTest.java
+++ b/shardingsphere-mode/shardingsphere-mode-core/src/test/java/org/apache/shardingsphere/mode/metadata/persist/service/impl/DataSourcePersistServiceTest.java
@@ -97,7 +97,6 @@ public final class DataSourcePersistServiceTest {
     
     private DataSource createDataSource(final String name) {
         MockedDataSource result = new MockedDataSource();
-        result.setDriverClassName("com.mysql.jdbc.Driver");
         result.setUrl("jdbc:mysql://localhost:3306/" + name);
         result.setUsername("root");
         result.setPassword("root");

--- a/shardingsphere-mode/shardingsphere-mode-core/src/test/resources/yaml/persist/data-source-foo.yaml
+++ b/shardingsphere-mode/shardingsphere-mode-core/src/test/resources/yaml/persist/data-source-foo.yaml
@@ -16,7 +16,6 @@
 #
 
 foo_ds:
-  driverClassName: com.mysql.jdbc.Driver
   password: root
   dataSourceClassName: org.apache.shardingsphere.test.mock.MockedDataSource
   connectionInitSqls:

